File: 0257.binary_tree_paths/solution.java
import java.util.ArrayList;

public class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class Solution {
    public List<String> binaryTreePaths(TreeNode root) {
        ArratList<String> paths = new ArrayList<>();
        String temp = "";
        deep(root, paths, temp);
        return paths;
    }

    public void deep(List<TreeNode> root, List<String> paths, String temp){
        if (root == null) return paths;
        temp += root.val;
        if (root.left == null && root.right == null){
            paths.add(temp);
        }
        deep(root.left, paths, temp+"->");
        deep(root.right, paths, temp+"->");
    }
}

File: 0257.binary_tree_paths/solution.py
from typing import List


from utils import TreeNode


class Solution:
    def binary_tree_paths(self, root: TreeNode) -> List[str]:
        paths = []

        def deep(root, path):
            if not root:
                return
            path = path + [str(root.val)]
            if not root.left and not root.right:
                return paths.append("->".join(path))
            deep(root.left, path)
            deep(root.right, path)

        deep(root, [])
        return paths
